File: 2015/day2/golden_star.py
f = open("input.txt", "r")
total = 0
i = 0
line = f.readline()
while line:
    line = line.split("x")
    l = int(line[0])
    w = int(line[1])
    h = int(line[2])
    total += w*h*l
    if l < w and l < h:
        total += 2*l
        if w < h:
            total += 2*w
        else:
            total += 2*h
    elif w < h:
        total += 2*w
        if l < h:
            total += 2*l
        else:
            total += 2*h
    else:
        total += 2*h
        if l < w:
            total += 2*l
        else:
            total += 2*w
    line = f.readline()


print(total)

File: 2023/day2/golden_star.java
package day2;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

public class golden_star {
 
    
    BufferedReader txt;

    public golden_star(File file) throws IOException{
        try{
            this.txt = new BufferedReader(new FileReader(file));
            System.out.println(this.fonction());
        }
        catch(FileNotFoundException e){
            System.out.println("File not found");
        }
    }

    public int fonction() throws IOException{
        int sum = 0;
        String line;
        while((line = this.txt.readLine()) != null){
            String[] split = line.split(":");
            int power = 0;
            String[] right = split[1].split(" ");
            int red = 0, green = 0, blue = 0;
            for (int i = 0; i+1 < right.length; i++) {
                if(right[i+1].contains("red") && right[i].matches("[0-9]+") && Integer.parseInt(right[i]) > red){
                    red = Integer.parseInt(right[i]);
                }else if(right[i+1].contains("green")  && right[i].matches("[0-9]+") && Integer.parseInt(right[i]) > green){
                    green = Integer.parseInt(right[i]);
                }else if(right[i+1].contains("blue")  && right[i].matches("[0-9]+") && Integer.parseInt(right[i]) >  blue){
                    blue = Integer.parseInt(right[i]);
                }
            }
            System.out.println(red + " " + green + " " + blue);
            power = red * green * blue;
            sum += power;
        }
        return sum;
    }


    public static void main(String[] args) throws IOException {
        File file = new File("day2/input.txt");
        golden_star day2_2 = new golden_star(file);
        
    }
}
